types/node/index.d.ts

```diff
@@ -1439,13 +1439,27 @@ declare module "zlib" {
         dictionary?: any; // deflate/inflate only, empty dictionary by default
     }
 
-    export interface Gzip extends stream.Transform { }
-    export interface Gunzip extends stream.Transform { }
-    export interface Deflate extends stream.Transform { }
-    export interface Inflate extends stream.Transform { }
-    export interface DeflateRaw extends stream.Transform { }
-    export interface InflateRaw extends stream.Transform { }
-    export interface Unzip extends stream.Transform { }
+    export interface Zlib {
+        readonly bytesRead: number;
+        close(callback?: Function): void;
+        flush(callback?: Function): void;
+    }
+
+    export interface ZlibParams {
+        params(level: number, strategy: number, callback?: Function): void;
```

Comment:
I think callback is mandatory here and a more concrete type would be `() => void`

Reply:
It's not mandatory (although it look like it in the documentation), and like above, we don't **know** the signature. I'm fine with `() => void` if you prefer it though.

Reply:
I was looking into the source and there the callback argument is directly passed to `process.nextTick()` without any if before (https://github.com/nodejs/node/blob/8a44289089a08b7b19fa3c4651b5f1f5d1edd71b/lib/zlib.js#L290).

This leads to my assumption that it is mandatory, no parameters are set and no return value is expected. Use of `() => void` instead `Function` just tells that you don't have to expect something useful nor someone is using the return value. Therefore I would prefer the stricter type - but I will not battle for it :wink:

Reply:
You're right. I looked around and the other two callbacks seem optional though. Done and pushed.